app/javascript/components/Admin/Products/Info/index.tsx

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { cast } from "ts-safe-cast";
+
+import { request } from "$app/utils/request";
+
+import AdminProductInfoContent from "$app/components/Admin/Products/Info/Content";
+import { type InfoProps } from "$app/components/Admin/Products/Info/Content";
+import { type Product } from "$app/components/Admin/Products/Product";
+import { useIsIntersecting } from "$app/components/useIsIntersecting";
+
+type Props = {
+  product: Product;
+};
+
+const AdminProductInfo = ({ product }: Props) => {
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [info, setInfo] = React.useState<InfoProps | null>(null);
+
+  const elementRef = useIsIntersecting<HTMLDivElement>((isIntersecting) => {
+    if (!isIntersecting || info) return;
+
+    const fetchInfo = async () => {
```

Comment:
You can probably leverage useLazyFetch here too, that will spare you from having to set `isLoading` yourself